### PR TITLE
IA-2432 add an unique constraint on form version

### PR DESCRIPTION
**Pull request by mathvdh**
Due to some concurrency issues, some clients managed to create Form Versions which have the same id even thou there is already a check about that when parsing the XLS file after upload via API.

Related JIRA tickets : IA-2432

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] My migrations file are included
- [x] Are there enough tests

## Changes

Adding of a unique_together DB contraint on FormVersion <form_id, version_id>

Adding a manual migration to check if some of those cases are there before adding the constraint and handling them by renaming them (increment the previous name).

## How to test

Not easy to test as it's protected at 2 levels (API and DB)
But see attached tests

### CODE REVIEW

**Comment by mathvdh** [comment]
@cypress

**Comment by mathvdh** [comment]
@cypress

**Comment by mathvdh** [comment]
@cypress

**Comment by mathvdh** [comment]
@cypress

**Comment by mathvdh** [comment]
@cypress

**Comment by mathvdh** [comment]
@cypress

File: iaso/migrations/0232_check_no_duplicated_version_id.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+            print("Updating other version to use version_id", vv.version_id, "\n")
+
+            # update corresponding instances
+
+            instances = Instance.objects.filter(form_version=vv)
+            for instance in instances:
+                print("Updating instance", instance.id, "for version", vv.version_id)
+
+                instance.json["_version"] = vv.version_id
```
Comment (madewulf) [inline iaso/migrations/0232_check_no_duplicated_version_id.py:55]: This won't fix the xml of the form, and I'm guessing that editing an instance with enketo would rewrite that field back to the version still in the xml

**Pull request closed without merging**